scripts/irods/test/test_iput.py

```diff
@@ -886,6 +887,50 @@ def test_large_put(self):
         self.object_name = 'test_large_put'
         self.run_tests(self.object_name, 5120000)
 
+    def test_iput_with_updated_checksum_read_buffer_size_in_bytes__issue_7947(self):
```

Comment:
Hmm, `test_iput.py` feels like the wrong place to put this. Given this change applies to multiple commands, we should find a python script that communicates that better.

We should add a test for `ichksum` to prove that the option applies to other commands too.

Reply:
I would suggest `test_icommands_file_operations` but I hate that file, so maybe each iCommand gets its own version of this test...

Reply:
After some discussion, we decided the following:
- The test must show that invalid values (e.g. negative integers) are detected and result in an error being returned to the client

> I would suggest `test_icommands_file_operations` but I hate that file, so maybe each iCommand gets its own version of this test...

That sounds reasonable to me.

Reply:
I added a test case to check invalid configuration options.  I have not added these for ichksum so leaving this unresolved for now.

Reply:
Added irepl thinking that that is what we want.  I will add ichksum.

Reply:
I added ichksum tests.  I also noticed that there was a library function to calculate the checksum in the format desired for the file itself so I modified iput/irepl tests to calculate the checksum on the file rather to writing the file to iRODS without server_config.json updates and then reading it.  I now just do the checksum comparison to what I calculated from the posix file.